src/Features/VisualBasic/Portable/CodeFixes/AddExplicitCast/VisualBasicAddExplicitCastCodeFixProvider.vb

```diff
@@ -0,0 +1,175 @@
+                        mutablePotentialConversionTypes.Add(Tuple.Create(spanNode, conversionType))
+                    End If
+                Case BC30518, BC30519
+                    Dim invocationNode = spanNode.GetAncestors(Of ExpressionSyntax).FirstOrDefault(
+                        Function(node) Not node.ChildNodes.OfType(Of ArgumentListSyntax).IsEmpty())
+                    Dim argumentList = invocationNode.ChildNodes.OfType(Of ArgumentListSyntax).FirstOrDefault()
+
+                    ' spanArgument is null because the span is on the invocation identifier name according to BC30519
+                    mutablePotentialConversionTypes.AddRange(GetPotentialConversionTypes(semanticModel, root,
+                        spanArgument:=Nothing, argumentList, invocationNode, cancellationToken))
+            End Select
+
+            ' clear up duplicate types
+            potentialConversionTypes = FilterValidPotentialConversionTypes(semanticModel, mutablePotentialConversionTypes)
+            Return Not potentialConversionTypes.IsEmpty
+        End Function
+
+        Protected Overrides Function IsObjectCreationExpression(targetNode As ExpressionSyntax) As Boolean
+            Return targetNode.Kind() = SyntaxKind.ObjectCreationExpression
+        End Function
+
+        Protected Overrides Function ClassifyConversionExists(semanticModel As SemanticModel,
+                expression As ExpressionSyntax, type As ITypeSymbol) As Boolean
+            Return semanticModel.ClassifyConversion(expression, type).Exists
+        End Function
+
+        Protected Overrides Function IsConversionIdentity(semanticModel As SemanticModel,
+                expression As ExpressionSyntax, type As ITypeSymbol) As Boolean
+            Return semanticModel.ClassifyConversion(expression, type).IsIdentity
+        End Function
+
+        Protected Overrides Function GetArguments(argumentList As SyntaxNode) As SeparatedSyntaxList(Of SyntaxNode)
+            Return If(TryCast(argumentList, ArgumentListSyntax)?.Arguments,
+                SyntaxFactory.SeparatedList(Of ArgumentSyntax)())
+        End Function
+
+        Protected Overrides Function GenerateNewArgument(
+                oldArgument As SyntaxNode, conversionType As ITypeSymbol) As SyntaxNode
+            Select Case oldArgument.Kind
+                Case SyntaxKind.SimpleArgument
+                    Dim simpleArgument = DirectCast(oldArgument, SimpleArgumentSyntax)
+                    Return simpleArgument.WithExpression(
+                        simpleArgument.GetExpression().Cast(conversionType, Nothing))
+                Case Else
+                    Return oldArgument
+            End Select
+        End Function
+
+        Protected Overrides Function GetArgumentExpression(argument As SyntaxNode) As ExpressionSyntax
+            Return TryCast(argument, ArgumentSyntax)?.GetExpression()
+        End Function
```

Comment:
already an ISyntaxFact.GetExpressionOfArgument.

Reply:
looks like this function doesn't support `AttributeArgumentSyntax` lol

Reply:
sounds like we're missing a test case for that then right?

Reply:
For `ISyntaxFact.GetExpressionOfArgument`? Since I add a attribute argument test to add explicit cast tests, I find that in C# there are `ArgumentSyntax` and `AttributeArgumentSyntax` and they are different(in VB there is only 'ArgumentSyntax'), thus I only use `SyntaxNode` as a generic type in methods. Is there any way I can simplify them?